mrp_client.py

```diff
@@ -69,6 +74,39 @@ def preseed(self, command, preseed_name, preseed_file, preseed_desc,
             self.log.fatal(err)
             exit(1)
 
+    def machine_control(self, machine_name, action, preseed_name, initrd_desc, kernel_desc,
+                             kernel_opts, arch, subarch):
+        try:
+            state = StateControl(self.urlhandler, machine_name)
+
+            if action == 'getparams':
+                machine_state = state.get_state()
+                print(self.__print_machine_state(machine_state))
+            else:
```

Comment:
Actions should be self-contained. Right now we only have two actions, so set_state should be inside "provision".
If, in the future, we have two or more actions that use exactly the same set_state, we can common them up by splitting the action if-chain:
```
if action1:
  ...
  return
elif action2:
  ...
  return
set_state()
if action3:
  ...
  return
elif action4:
  ...
  return
...
```